lib/matplotlib/figure.py

```diff
@@ -364,7 +365,7 @@ def __init__(self,
         self._set_artist_props(self.patch)
         self.patch.set_antialiased(False)
 
-        self.canvas = None
+        FigureCanvasBase(self)  # set .canvas.
```

Comment:
The issue I am having with this idea is that with the convoluted Figure/FigureManager/Canvas relationships across backends, is the right subclassed Canvas being assigned? What about for subclassed Figures?

Reply:
The backend (if using pyplot) will swap in the correct canvas later (as it always did).
We don't subclass Figures (I think? (and even then, it's not clear how this would be a problem)).

Reply:
I would prefer we still have a `self.canvas = None` line above this for clarity

Reply:
Fair point on the type checker.  

One of my code exploration trick is to grep for `self.ATTR` so it is a useful breadcrumb to leave.  Don't feel that strongly.

Reply:
Edited the comment so that your grep will match it :p